drivers/power/ltc7841/ltc7841.h

```diff
@@ -0,0 +1,189 @@
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef __LTC7841_H__
+#define __LTC7841_H__
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include "no_os_i2c.h"
+#include "no_os_irq.h"
+#include "no_os_print_log.h"
+#include "no_os_delay.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_units.h"
+#include "no_os_alloc.h"
+
+/** X is set based on the pull configuration of the ADDR pin */
+#define TWO_BYTES_LENGTH            2
+#define LTC7841_ADDR                (0x20)
+#define REPEATED_START_MODE         1
+#define I2C_WRITE_LENGTH_FOR_READ   1
+#define I2C_DEFAULT_ARRAY_SIZE      2
+#define STOP_MODE                   0
+#define LTC7841_DEFAULT_ADDRESS_0   0x5A
+#define LTC7841_DEFAULT_ADDRESS_1   0x5B
+#define LTC7841_TOTAL_REGISTERS     0x16
+#define LTC7841_BYTE_LENGTH         1
+#define LTC7841_WORD_LENGTH         2
+#define LTC7841_I2C_WR_FRAME_SIZE   3
+/* this denotes the rsense scale of LTC7841 2(both channels) * 50 uV/ bit / 0.002 (rsense resistor), unit in uA*/
+#define CURRENT_SCALE               (20)
+/*this denotes the max voltage output of LTC7841 in millivolts*/
+#define LTC7841_MAX_VOUT            48000
+#define LTC7841_7_BIT_ADDRESS_MASK  0x7F
+#define LTC7841_OPERATION_OFF       0x00
+#define LTC7841_OPERATION_TURN_ON   0x80
+#define LTC7841_OPERATION_MARGIN_LOW 0x98
+#define LTC7841_OPERATION_MARGIN_HIGH 0xA8
+/*slew rate multiplier*/
+#define LTC7841_NOMINAL_SLEW_RATE   0b00
+#define LTC7841_SLOW_SLEW_RATE      0b01
+#define LTC7841_FAST_SLEW_RATE      0b11
+/*READ TEMPERATURE 1*/
+#define LTC7841_INTERNAL_DIE_TEMPERATURE 0b0
+#define LTC7841_TSNS_PIN_VOLTAGE         0b1
+#define READ_TEMPERATURE_1_BIT      2
+/*MFR MARGIN MAX VALUE*/
+#define MFR_MAX_VALUE               0x1FF
+/*MFR VOUT INITIALIZE TO HALF */
+#define MFR_VOUT_INIT              0x0FF
+#define WRITE_ONLY_REGISTERS_NUMBER 2
+#define ECOMM                       9
+/* specific only to DC2798A in ohms*/
+#define RSENSE_VALUE_IIN            (500)
```

Comment:
This might be something you will need to add as param in the example application. Anything that is specific to the board level (i.e. outside the chip) add it as param. However, this is not used anywhere in you driver and iio_driver.